docs/tutorials/data_provider.md

```diff
@@ -311,3 +311,62 @@ let formatter = FixedDecimalFormatter::try_new_with_any_provider(
 
 assert_eq!(formatter.format_to_string(&100007i64.into()), "100🐮007");
 ```
+
+## Accessing the Resolved Locale
+
+ICU4X objects do not store their "resolved locale" because that is not a well-defined concept. Components can load data from many sources, and fallbacks to parent locales or root does not necesarily mean that a locale is not supported.
+
+However, for environments that require this behavior, such as ECMA-402, the data provider can be instrumented to access the resolved locale from `DataResponseMetadata`, as shown in the following example.
+
+```rust
+use icu_provider::prelude::*;
+use icu_provider::hello_world::*;
+use icu::decimal::FixedDecimalFormatter;
+use icu_provider_adapters::fallback::LocaleFallbackProvider;
+use icu_provider_adapters::fallback::LocaleFallbacker;
+use icu::locid::locale;
+use std::sync::RwLock;
+
+pub struct ResolvedLocaleProvider<P> {
+    inner: P,
+    resolved_locale: RwLock<Option<DataLocale>>,
+}
+
+impl<P> AnyProvider for ResolvedLocaleProvider<P>
+where
+    P: AnyProvider
+{
+    fn load_any(&self, key: DataKey, req: DataRequest) -> Result<AnyResponse, DataError> {
+        let mut any_res = self.inner.load_any(key, req)?;
+        // Whichever locale gets loaded for `HelloWorldV1Marker::KEY` will be the one
+        // we consider the "resolved locale".
+        if key == HelloWorldV1Marker::KEY {
```

Comment:
kind of pointless because the provider only supports that one key

Reply:
Yeah I wanted to demonstrate how this works with a constructor that loads multiple keys but we're restricted to HelloWorldProvider. However I think it's still important to illustrate that you should be sniffing for a particular key.

Reply:
Added some docs.